src/QtLocationPlugin/QGeoTileFetcherQGC.cpp

```diff
@@ -1,82 +1,86 @@
-** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
-**
-** In addition, as a special exception, Digia gives you certain additional
-** rights.  These rights are described in the Digia Qt LGPL Exception
-** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
-**
-** GNU General Public License Usage
-** Alternatively, this file may be used under the terms of the GNU
-** General Public License version 3.0 as published by the Free Software
-** Foundation and appearing in the file LICENSE.GPL included in the
-** packaging of this file.  Please review the following information to
-** ensure the GNU General Public License version 3.0 requirements will be
-** met: http://www.gnu.org/copyleft/gpl.html.
-**
-**
-** $QT_END_LICENSE$
-**
-** 2015.4.4
-** Adapted for use with QGroundControl
-**
-** Gus Grubba <gus@auterion.com>
-**
-****************************************************************************/
-
 #include "QGeoTileFetcherQGC.h"
+#include "QGeoTiledMappingManagerEngineQGC.h"
 #include "QGCMapEngine.h"
 #include "QGeoMapReplyQGC.h"
 #include "QGCMapUrlEngine.h"
+#include "MapProvider.h"
+#include <DeviceInfo.h>
+#include <QGCLoggingCategory.h>
 
 #include <QtNetwork/QNetworkRequest>
 #include <QtLocation/private/qgeotilespec_p.h>
 #include <QtLocation/private/qgeotiledmappingmanagerengine_p.h>
 
-//-----------------------------------------------------------------------------
-QGeoTileFetcherQGC::QGeoTileFetcherQGC(QGeoTiledMappingManagerEngine *parent)
+QGC_LOGGING_CATEGORY(QGeoTileFetcherQGCLog, "qgc.qtlocationplugin.qgeotilefetcherqgc")
+
+QGeoTileFetcherQGC::QGeoTileFetcherQGC(QNetworkAccessManager *networkManager, QGeoTiledMappingManagerEngineQGC *parent)
     : QGeoTileFetcher(parent)
-    , _networkManager(new QNetworkAccessManager(this))
+    , m_networkManager(networkManager)
 {
-
+    // qCDebug(QGeoTileFetcherQGCLog) << Q_FUNC_INFO << this;
 }
 
-//-----------------------------------------------------------------------------
 QGeoTileFetcherQGC::~QGeoTileFetcherQGC()
 {
-
+    // qCDebug(QGeoTileFetcherQGCLog) << Q_FUNC_INFO << this;
 }
 
-//-----------------------------------------------------------------------------
-QGeoTiledMapReply*
-QGeoTileFetcherQGC::getTileImage(const QGeoTileSpec &spec)
+QGeoTiledMapReply* QGeoTileFetcherQGC::getTileImage(const QGeoTileSpec &spec)
 {
-    //-- Build URL
-    QNetworkRequest request = UrlFactory::getTileURL(spec.mapId(), spec.x(), spec.y(), spec.zoom());
-    if ( ! request.url().isEmpty() ) {
-        return new QGeoTiledMapReplyQGC(_networkManager, request, spec);
+    const QList<SharedMapProvider> providers = UrlFactory::getProviders();
+    int id = spec.mapId();
+    if ((id < 1) || (id > providers.size())) {
+        qCWarning(QGeoTileFetcherQGCLog) << QString("Unknown map id %1").arg(id);
+        if (providers.isEmpty()) {
+            return nullptr;
+        } else {
+            id = 1;
+        }
     }
```

Comment:
Instead of this, wouldn't it be better to just check if `provider` is null below?